### PR TITLE
hw05  first part

### PR DESCRIPTION
**Pull request by OmarTayyar**
half of hw05 is ready except test classes

### CODE REVIEW

File: src/main/java/az/edu/strangers/DomesticCat.java [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
+
 public class DomesticCat extends Pet implements Foulable {
 
-    public DomesticCat(String nickName, Integer age, Integer trickLevel, String[] habits) {
-        super(nickName, age, trickLevel, habits);
+    public DomesticCat(String nickName, Integer age, Integer trickLevel, Set<String> habits) {
+        this.nickName = nickName;
+        this.age = age;
+        this.trickLevel = trickLevel;
+        this.habits = habits;
```
Comment (Shammad-Kosayev) [inline src/main/java/az/edu/strangers/DomesticCat.java:11]: revert code back to `super(nickname, age, tricklevel, habits)`

Comment (thapdigov) [reply src/main/java/az/edu/strangers/DomesticCat.java:11]: yes I wanted to  this

File: src/main/java/az/edu/strangers/Human.java [hunk 2]
```diff
@@ -100,21 +101,21 @@ public void describePet() {
     }
 
     public String toString() {
-        return "Human{name='%s', surname='%s', year=%d, iq=%s, schedule=%s}".formatted(name, surname, year, IQ, Arrays.deepToString(schedule));
+        return "Human{name='%s', surname='%s', year=%d, iq=%s, schedule=%s}".formatted(name, surname, year, IQ, Objects.toString(schedule));
```
Comment (thapdigov) [inline src/main/java/az/edu/strangers/Human.java:104]: There is no need to call with the object. It is enough to write getSchedule.

File: src/main/java/az/edu/strangers/Pet.java [hunk 3]
```diff
@@ -68,17 +71,17 @@ public void setTrickLevel(Integer trickLevel) {
...
 
-    public void setHabits(String[] habits) {
+    public void setHabits(Set<String> habits) {
         this.habits = habits;
     }
 
     public String toString() {
         return "%s{nickname='%s', age=%d, trickLevel=%d, habits=%s}"
-                .formatted(species, nickName, age, trickLevel, Arrays.toString(habits));
+                .formatted(species, nickName, age, trickLevel, habits);
```
Comment (thapdigov) [inline src/main/java/az/edu/strangers/Pet.java:84]: does habits work? I think getHabits is more suitable

File: src/test/java/az/edu/strangers/FamilyTest.java [hunk 4]
```diff
@@ -18,10 +23,15 @@ void setUp() {
         mother = new Human("Jane", "Karleone", 1975);
         father = new Human("Vito", "Karleone", 1973);
         family = new Family(mother, father);
-        pet = new Pet(Species.DOG, "Rock", 5, 75, new String[]{"eat", "sleep"});
+        Set<String> habits = new HashSet<>();
+        habits.add("Sleep");
+        habits.add("Run");
+        pet = new Pet(Species.DOG, "Rock", 5, 75, new String[] {"eat" , "sleep"});
```
Comment (thapdigov) [inline src/test/java/az/edu/strangers/FamilyTest.java:29]: Why array?

File: src/main/java/az/edu/strangers/Pet.java (hunk 3 above)
Comment (Shammad-Kosayev) [reply src/main/java/az/edu/strangers/Pet.java:84]: habits is in Pet class. why bother call getHabits when you can directly access habits?

File: src/test/java/az/edu/strangers/FamilyTest.java (hunk 4 above)
Comment (Shammad-Kosayev) [reply src/test/java/az/edu/strangers/FamilyTest.java:29]: Test classes will be reverted back, i will handle them.

**Review by Shammad-Kosayev — CHANGES_REQUESTED** [review]
Changes required to be implemented

File: src/main/java/az/edu/strangers/Pet.java (hunk 3 above)
Comment (thapdigov) [reply src/main/java/az/edu/strangers/Pet.java:84]: it's just a habit:))

**Comment by OmarTayyar** [comment]
nanay

**Pull request closed without merging**